src/flamegpu/gpu/CUDAAgent.cu

```diff
@@ -24,8 +24,11 @@
 #include "flamegpu/gpu/CUDAScatter.h"
 
 #ifdef _MSC_VER
-#pragma warning(push, 3)
+#pragma warning(push, 0)
+#pragma warning(disable : 4706)
+// #pragma diag_suppress = 1719  // Doesn't work at suppressing wreorder
```

Comment:
Are these comments supposed to be left in?
Are they not relevant at all the other cub includes too?

Reply:
I semi-intentionally left the commented out `#pragma diag_suppress` for future reference, though it could be removed. 

The `#pragma warning(disable : 4706)` can probably be removed too now I've set the warning level to 0, though it also doesn't cause any harm.